daemon/cmd/daemon.go

```diff
@@ -216,7 +217,9 @@ type Daemon struct {
 	// just used to tie together some status reporting
 	cniConfigManager cni.CNIConfigManager
 
-	l2announcer *l2announcer.L2Announcer
+	l2announcer    *l2announcer.L2Announcer
+	healthProvider cell.Health
+	healthReporter cell.HealthReporter
```

Comment:
Speaking of logical separations, what do you think of combining the reporter into the provider? Do we really need two separate cells? At a surface level, it seems reasonable that a health provider has a "reporter" component.